cider/tests/functionality/CiderGroupByTest.cpp

```diff
@@ -48,6 +48,29 @@
     }                                                                                  \
   };
 
+class CiderGroupByVarcharTest : public CiderTestBase {
+ public:
+  // TODO(yizhong): make col_d to 50% chance nullable and change min length to 0
+  // after null string is supported
+  CiderGroupByVarcharTest() {
+    table_name_ = "table_test";
+    create_ddl_ =
+        "CREATE TABLE table_test(col_a BIGINT NOT NULL, col_b BIGINT NOT NULL, col_c "
+        "VARCHAR NOT NULL, col_d VARCHAR);";
+    input_ = {std::make_shared<CiderBatch>(
+        QueryDataGenerator::generateBatchByTypes(500,
+                                                 {"col_a", "col_b", "col_c", "col_d"},
+                                                 {CREATE_SUBSTRAIT_TYPE(I64),
+                                                  CREATE_SUBSTRAIT_TYPE(I64),
+                                                  CREATE_SUBSTRAIT_TYPE(Varchar),
+                                                  CREATE_SUBSTRAIT_TYPE(Varchar)},
+                                                 {0, 0, 0, 0},
```

Comment:
Don't you need set correct null_chance based on your defined schema?

Reply:
Commented a TODO at Line53, let me change the `null_chance` after null string is supported.